src/components/contacts/components/OTPInput.tsx

```diff
@@ -0,0 +1,139 @@
+import {useRef, useState} from 'react'
+import {
+  Pressable,
+  TextInput,
+  type TextInputSelectionChangeEvent,
+  View,
+} from 'react-native'
+import {msg} from '@lingui/macro'
+import {useLingui} from '@lingui/react'
+
+import {mergeRefs} from '#/lib/merge-refs'
+import {isIOS} from '#/platform/detection'
+import {atoms as a, platform, useTheme} from '#/alf'
+import {useInteractionState} from '#/components/hooks/useInteractionState'
+import {Text} from '#/components/Typography'
+
+export function OTPInput({
+  label,
+  value,
+  onChange,
+  ref,
+  numberOfDigits = 6,
+  onComplete,
+}: {
+  label: string
+  value: string
+  onChange: (text: string) => void
+  ref?: React.Ref<TextInput>
+  numberOfDigits?: number
+  onComplete?: (code: string) => void
+}) {
+  const t = useTheme()
+  const {_} = useLingui()
+  const innerRef = useRef<TextInput>(null)
+  const {state: focused, onIn: onFocus, onOut: onBlur} = useInteractionState()
+  const [selection, setSelection] = useState({start: 0, end: 0})
+
+  const onChangeText = (text: string) => {
+    // only numbers
+    text = text.replace(/[^0-9]/g, '')
+    text = text.slice(0, numberOfDigits)
+    onChange(text)
+    if (text.length === numberOfDigits) {
+      onComplete?.(text)
+      innerRef.current?.blur()
+    }
+  }
+
+  const onSelectionChange = (evt: TextInputSelectionChangeEvent) => {
+    setSelection(evt.nativeEvent.selection)
+  }
+
+  return (
+    <Pressable
+      accessibilityLabel={_(msg`Focus OTP input`)}
```

Comment:
```suggestion
      accessibilityLabel={_(msg`Focus code input`)}
```
nit: `OTP` might not be a term that many users are familiar with, `code` might work better